src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -3354,15 +3354,22 @@ void SmartScript::UpdateTimer(SmartScriptHolder& e, uint32 const diff)
                 if (e.GetScriptType() == SMART_SCRIPT_TYPE_TIMED_ACTIONLIST)
                 {
                     e.enableTimed = false;//disable event if it is in an ActionList and was processed once
+                    bool isActionListOver = true;
+
                     for (SmartAIEventList::iterator i = mTimedActionList.begin(); i != mTimedActionList.end(); ++i)
                     {
                         //find the first event which is not the current one and enable it
                         if (i->event_id > e.event_id)
                         {
                             i->enableTimed = true;
+                            isActionListOver = false;
                             break;
                         }
                     }
+
+                    // Set object to non-active if there's nothing else to process
+                    if (isActionListOver)
+                        GetBaseObject()->setActive(false);
```

Comment:
I wonder about manual setActive() calls, this will set it to false even if unintended. It should restore the previous value (but that complicates things a bit)

Reply:
@jackpoz is this acceptable? https://pastebin.com/CAvrscB8